src/project.ts

```diff
@@ -132,6 +143,62 @@ export async function readProject(
   };
 }
 
+/**
+ * Returns the https-prefixed GitHub repository URL for the project, so that we
+ * know how to construct links to releases inside of the changelog.
+ *
+ * This URL is obtained by looking at the following sources, in this order:
+ *
+ * - `process.env.npm_package_repository_url` (to support MetaMask projects
+ * which still use NPM or Yarn 1.x)
+ * - The `repository` field inside of `package.json` (to support most MetaMask
+ * projects)
+ * - The URL of the "origin" remote (to support newer projects which for some
+ * reason do not have a "repository" field)
+ *
+ * Note that there is a function in `@metamask/auto-changelog` that offers a
+ * similar capability, and in fact, this function should match it as closely as
+ * possible so that when changelogs are updated they also pass
+ * `@metamask/auto-changelog`'s validation step.
+ *
+ * @param packageManifest - The manifest for the package that represents a
+ * project.
+ * @param repositoryDirectoryPath - The path to the repository for the project.
+ * @returns The HTTPS URL of the repository, e.g.
+ * `https://github.com/OrganizationName/RepositoryName`.
+ * @throws If the URL obtained via `package.json` or the "origin" remote is in
+ * in invalid format.
+ */
+export async function getValidRepositoryUrl(
+  packageManifest: UnvalidatedPackageManifest,
+  repositoryDirectoryPath: string,
+): Promise<string> {
+  // Set automatically by NPM or Yarn 1.x
+  const npmPackageRepositoryUrl = process.env.npm_package_repository_url;
+
+  if (npmPackageRepositoryUrl) {
+    return convertToHttpsGitHubRepositoryUrl(npmPackageRepositoryUrl);
+  }
+
+  if (typeof packageManifest.repository === 'string') {
+    return convertToHttpsGitHubRepositoryUrl(packageManifest.repository);
+  }
+
+  if (
+    isPlainObject(packageManifest.repository) &&
+    typeof packageManifest.repository.url === 'string'
+  ) {
+    return convertToHttpsGitHubRepositoryUrl(packageManifest.repository.url);
+  }
+
+  const gitConfigUrl = await getStdoutFromCommand(
```

Comment:
Given that most of our projects ought to have a `repository` field in `package.json`, I am not sure that this is really needed, but I figured I'd leave it here.

I tracked down the original source for this logic, and it comes from here: https://github.com/MetaMask/action-monorepo-release-pr/commit/af8d7504aa7b8cd4a65a81625839adf3afc3cbb9#diff-b2d6c9ae0590887f3d213032d057be61bd3e628650fb7ba99391bdc84587e454R25